identity/test/controllers/ConsentsJourneyControllerTest.scala

```diff
@@ -0,0 +1,219 @@
+    val idUrlBuilder = mock[IdentityUrlBuilder]
+    val api = mock[IdApiClient]
+    val idRequestParser = mock[IdRequestParser]
+    val authService = mock[AuthenticationService]
+    val idRequest = mock[IdentityRequest]
+    val trackingData = mock[TrackingData]
+    val returnUrlVerifier = mock[ReturnUrlVerifier]
+    val newsletterService = spy(new NewsletterService(api, idRequestParser, idUrlBuilder))
+    val httpConfiguration = HttpConfiguration.createWithDefaults()
+
+    val userId: String = "123"
+    val user = User("test@example.com", userId, statusFields = StatusFields(receive3rdPartyMarketing = Some(true), receiveGnmMarketing = Some(true), userEmailValidated = Some(true)))
+    val testAuth = ScGuU("abc", GuUCookieData(user, 0, None))
+    val authenticatedUser = AuthenticatedUser(user, testAuth, true)
+    val phoneNumbers = PhoneNumbers
+
+    val redirectDecisionService = new ProfileRedirectService(newsletterService, idRequestParser, controllerComponent)
+    val authenticatedActions = new AuthenticatedActions(authService, api, mock[IdentityUrlBuilder], controllerComponent, newsletterService, idRequestParser, redirectDecisionService)
+
+    val profileFormsMapping = ProfileFormsMapping(
+      new AccountDetailsMapping,
+      new PrivacyMapping,
+      new ProfileMapping
+    )
+
+    when(authService.fullyAuthenticatedUser(MockitoMatchers.any[RequestHeader])) thenReturn Some(authenticatedUser)
+    when(api.me(testAuth)) thenReturn Future.successful(Right(user))
+
+    when(idRequest.trackingData) thenReturn trackingData
+    when(idRequest.returnUrl) thenReturn None
+    when(idRequestParser.apply(MockitoMatchers.any[RequestHeader])) thenReturn idRequest
+
+    when(returnUrlVerifier.defaultReturnUrl) thenReturn "http://1234.67"
+    when(returnUrlVerifier.getVerifiedReturnUrl(MockitoMatchers.any[RequestHeader])) thenReturn None
+    when(api.userEmails(MockitoMatchers.anyString(), MockitoMatchers.any[TrackingData])) thenReturn Future.successful(Right(Subscriber("Text", List(EmailList("37")))))
+    when(api.updateUserEmails(MockitoMatchers.anyString(), MockitoMatchers.any[Subscriber], MockitoMatchers.any[Auth], MockitoMatchers.any[TrackingData])) thenReturn Future.successful(Right(()))
+
+    lazy val controller = new EditProfileController(
+      idUrlBuilder,
+      redirectDecisionService,
+      authenticatedActions,
+      api,
+      idRequestParser,
+      csrfCheck,
+      csrfAddToken,
+      returnUrlVerifier,
+      newsletterService,
+      profileFormsMapping,
+      testApplicationContext,
+      httpConfiguration,
+      controllerComponent
+    )
+  }
+
+  "ConsentsJourney" when {
+
+    "using any journey" should {
+
+      "have a js fallback" in new ConsentsJourneyFixture {
+        val result = controller.displayConsentsJourney(None).apply(FakeCSRFRequest(csrfAddToken))
+        status(result) should be(200)
+        contentAsString(result) should include ("consents : navigation : submit-force")
```

Comment:
generally I'm not a big fan of tests that use `contentAsString` but I guess there isn't much alternative in these cases?

Reply:
we do import JSoup into frontend, which we could use for parsing the result? might be overkill (and heavyweight) though

Reply:
I had no idea we had such a thing! Might be too much for this PR but I'll keep it in mind for the future